File: Android/App/app/src/main/java/br/ufscar/dc/mobile/app/model/db/CategoryDao.kt
package br.ufscar.dc.mobile.app.model.db

import androidx.lifecycle.LiveData
import androidx.room.*
import br.ufscar.dc.mobile.app.model.Category

@Dao
interface CategoryDao {
    @Query("SELECT * FROM Category")
    suspend fun getAll(): List<Category>

    @Query("SELECT * FROM Category WHERE id == :id")
    suspend fun get(id: String): Category

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(category: Category)
}

File: Flutter/mobile_trabalho/lib/db/dao/categoryDao.dart
import 'package:floor/floor.dart';
import 'package:mobile_trabalho/db/entity/category.dart';

@dao
abstract class CategoryDao {
  @Query("SELECT * FROM Category")
  Future<List<Category>> getAll();

  @Query("SELECT * FROM Category WHERE id == :id")
  Future<Category?> findCategoryById(String id);

  @insert
  Future<void> insertCategory(Category category);
}
